graphql_compiler/tests/test_compiler.py

```diff
@@ -2868,3 +2868,907 @@ def test_traverse_and_optional_and_traverse(self):
         '''
 
         check_test_data(self, test_data, expected_match, expected_gremlin)
+
+    def test_multiple_optional_traversals_with_starting_filter(self):
+        test_data = test_input_data.multiple_optional_traversals_with_starting_filter()
+
+        expected_match = '''
+            SELECT EXPAND($result)
+            LET
+            $optional__0 = (
+                SELECT Animal___1.name AS `animal_name`
+                    FROM (
+                        MATCH {{
+                            class: Animal,
+                            where: ((
+                                (
+                                    (out_Animal_ParentOf IS null)
+                                    OR
+                                    (out_Animal_ParentOf.size() = 0)
+                                )
+                                AND
+                                (
+                                    (name LIKE ('%' + ({wanted} + '%')))
+                                    AND
+                                    (
+                                        (in_Animal_ParentOf IS null)
+                                        OR
+                                        (in_Animal_ParentOf.size() = 0)
+                                    )
+                                )
+                            )),
+                            as: Animal___1
+                        }} ,
+                        {{
+                            class: Animal,
+                            as: Animal___1
+                        }}
+                        RETURN $matches
+                    )
+            ),
+            $optional__1 = (
+                SELECT Animal___1.name AS `animal_name`,
+                        Animal__in_Animal_ParentOf___1.name AS `parent_name`,
```

Comment:
bad indentation, and also doesn't follow the style of other MATCH queries in this file